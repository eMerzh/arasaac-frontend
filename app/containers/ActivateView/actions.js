/*
 *
 * LoginView actions
 *
 */

import { createRequestTypes, action } from 'utils/actions'

// constants
export const ACTIVATION = createRequestTypes('ACTIVATION')

export const activation = {
  request: (code) => action(ACTIVATION.REQUEST, { code }),
  success: () => action(ACTIVATION.SUCCESS),
  failure: (error) => action(ACTIVATION.FAILURE, { error })
}
