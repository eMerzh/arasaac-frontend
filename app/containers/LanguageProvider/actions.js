/*
 *
 * LanguageProvider actions
 *
 */

export const CHANGE_LOCALE = 'app/LanguageToggle/CHANGE_LOCALE'
export const START_TRANSLATION = 'app/LanguageToggle/START_TRANSLATION'
export const STOP_TRANSLATION = 'app/LanguageToggle/STOP_TRANSLATION'
export const changeLocale = (languageLocale) => ({ type: CHANGE_LOCALE, locale: languageLocale })
export const startTranslation = () => ({ type: START_TRANSLATION })
export const stopTranslation = () => ({ type: STOP_TRANSLATION })
