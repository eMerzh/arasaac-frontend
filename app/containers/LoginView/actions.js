// actions moved to App/actions
