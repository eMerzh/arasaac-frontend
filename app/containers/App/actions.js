/*
 *
 * LoginView actions
 *
 */

import { createRequestTypes, action } from "utils/actions";
import { DEFAULT_LIST } from "utils";

// constants
export const LOGIN = createRequestTypes("LOGIN");
export const SOCIAL_LOGIN = createRequestTypes("SOCIAL_LOGIN");
export const LOGOUT = "app/LoginView/LOGOUT";
export const TOKEN_VALIDATION = createRequestTypes("TOKEN_VALIDATION");
export const TOKEN_REFRESH = createRequestTypes("TOKEN_REFRESH");
export const RESET_ERROR = "app/LoginView/RESET_ERROR";
export const ACTIVATION = createRequestTypes("ACTIVATION");
// for profileView
export const UPDATE_USER = createRequestTypes("UPDATE_USER");

// pictograms, saga takes place in PictogramsView
export const ADD_FAVORITE = createRequestTypes("ADD_FAVORITE");
export const DELETE_FAVORITE = createRequestTypes("DELETE_FAVORITE");
export const ADD_LIST = createRequestTypes("ADD_LIST");
export const RENAME_LIST = createRequestTypes("RENAME_LIST");
export const DELETE_LIST = createRequestTypes("DELETE_LIST");

export const login = {
  request: (username, password) =>
    action(LOGIN.REQUEST, { username, password }),
  success: (accessToken, refreshToken) =>
    action(LOGIN.SUCCESS, { accessToken, refreshToken }),
  failure: (error) => action(LOGIN.FAILURE, { error }),
};

export const socialLogin = {
  request: (socialToken, provider, locale) =>
    action(SOCIAL_LOGIN.REQUEST, { socialToken, provider, locale }),
  success: (accessToken, refreshToken) =>
    action(SOCIAL_LOGIN.SUCCESS, { accessToken, refreshToken }),
  failure: (error) => action(SOCIAL_LOGIN.FAILURE, { error }),
};

export const tokenValidation = {
  request: (accessToken) => action(TOKEN_VALIDATION.REQUEST, { accessToken }),
  success: (authData) => action(TOKEN_VALIDATION.SUCCESS, { authData }),
  failure: (error) => action(TOKEN_VALIDATION.FAILURE, { error }),
};

export const tokenRefresh = {
  request: (refreshToken) => action(TOKEN_REFRESH.REQUEST, { refreshToken }),
  success: (accessToken) => action(TOKEN_REFRESH.SUCCESS, { accessToken }),
  failure: (error) => action(TOKEN_REFRESH.FAILURE, { error }),
};

export const activation = {
  request: (code) => action(ACTIVATION.REQUEST, { code }),
  success: (accessToken) => action(ACTIVATION.SUCCESS, { accessToken }),
  failure: (error) => action(ACTIVATION.FAILURE, { error }),
};

export const updateUser = {
  request: (user, token) => action(UPDATE_USER.REQUEST, { user, token }),
  success: (user) => action(UPDATE_USER.SUCCESS, { user }),
  failure: (error) => action(UPDATE_USER.FAILURE, { error }),
};

export const logout = () => action(LOGOUT);

export const resetError = () => action(RESET_ERROR);

// favorite pictograms related actions:

export const addFavorite = {
  request: (fileName, listName = DEFAULT_LIST, token) =>
    action(ADD_FAVORITE.REQUEST, { fileName, listName, token }),
  success: (fileName, listName) =>
    action(ADD_FAVORITE.SUCCESS, { fileName, listName }),
  failure: (error) => action(ADD_FAVORITE.FAILURE, { error }),
};

export const deleteFavorite = {
  request: (fileName, listName = DEFAULT_LIST, token) =>
    action(DELETE_FAVORITE.REQUEST, { fileName, listName, token }),
  success: (fileName, listName) =>
    action(DELETE_FAVORITE.SUCCESS, { fileName, listName }),
  failure: (error) => action(DELETE_FAVORITE.FAILURE, { error }),
};

export const addList = {
  request: (listName, token) => action(ADD_LIST.REQUEST, { listName, token }),
  success: (listName) => action(ADD_LIST.SUCCESS, { listName }),
  failure: (error) => action(ADD_LIST.FAILURE, { error }),
};

export const renameList = {
  request: (listName, newListName, token) =>
    action(RENAME_LIST.REQUEST, { listName, newListName, token }),
  success: (listName, newListName) =>
    action(RENAME_LIST.SUCCESS, { listName, newListName }),
  failure: (error) => action(RENAME_LIST.FAILURE, { error }),
};

export const deleteList = {
  request: (listName, token) =>
    action(DELETE_LIST.REQUEST, { listName, token }),
  success: (listName) => action(DELETE_LIST.SUCCESS, { listName }),
  failure: (error) => action(DELETE_LIST.FAILURE, { error }),
};
