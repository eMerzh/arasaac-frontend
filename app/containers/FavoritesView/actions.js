/*
 *
 * PictogramsView actions
 *
 */

import { createRequestTypes, action } from 'utils/actions'

// constants
export const FAVORITE_PICTOGRAMS = createRequestTypes('FAVORITE_PICTOGRAMS')
export const FAVORITE_LIST_SELECT = 'FAVORITE_LIST_SELECT'

export const favoriteListSelect = (listName) =>
  action(FAVORITE_LIST_SELECT, { listName })

export const favoritePictograms = {
  request: (locale, favoriteIds, token) =>
    action(FAVORITE_PICTOGRAMS.REQUEST, { locale, favoriteIds, token }),
  success: (locale, data) =>
    action(FAVORITE_PICTOGRAMS.SUCCESS, { locale, data }),
  failure: (error) => action(FAVORITE_PICTOGRAMS.FAILURE, { error })
}
