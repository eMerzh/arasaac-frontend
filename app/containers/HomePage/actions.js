import { createRequestTypes, action } from 'utils/actions'
import { DEFAULT_LIST } from 'utils'

// constants
export const START_TOUR = 'START_TOUR'
export const STOP_TOUR = 'STOP_TOUR'

export const startTour = () => action(START_TOUR)

export const stopTour = () => action(STOP_TOUR)