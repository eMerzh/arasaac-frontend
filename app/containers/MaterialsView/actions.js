/*
 *
 * PictogramsView actions
 *
 */

import { createRequestTypes, action } from 'utils/actions'

// constants
export const MATERIALS = createRequestTypes('app/MaterialsView/MATERIALS')
export const SHOW_FILTERS = 'app/MaterialsView/SHOW_FILTERS'

// actions: pictograms.request/success/failure

export const materials = {
  request: (searchText) => action(MATERIALS.REQUEST, { searchText }),
  success: (searchText, data) => action(MATERIALS.SUCCESS, { searchText, data }),
  failure: (error) => action(MATERIALS.FAILURE, { error })
}

export const toggleShowFilter = () => action(SHOW_FILTERS)
