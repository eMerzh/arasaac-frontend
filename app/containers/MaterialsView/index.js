/*
 *
 * MaterialsView
 *
 */
import React, { PropTypes } from 'react'
import { connect } from 'react-redux'
import { FormattedMessage } from 'react-intl'
import View from 'components/View'
import Helmet from 'react-helmet'
import SearchField from 'components/SearchField'
import Toggle from 'material-ui/Toggle'
import { Map } from 'immutable'
import FilterList from 'components/Filters'
import { withRouter } from 'react-router'
import { denormalize } from 'normalizr'
import { searchMaterialSchema } from 'services/schemas'
import { getFilteredItems } from 'utils'
import { materials, toggleShowFilter, setFilterItems } from './actions'
import messages from './messages'

class MaterialsView extends React.Component { // eslint-disable-line react/prefer-stateless-function

  componentDidMount() {
    if (this.props.params.searchText) {
      this.props.requestMaterials(this.props.locale, this.props.params.searchText)
    }
  }
  componentWillReceiveProps(nextProps) {
    if (this.props.params.searchText !== nextProps.params.searchText) {
      this.props.requestMaterials(this.props.locale, nextProps.params.searchText)
    }
  }

  handleSubmit = (nextValue) => {
    if (this.props.params.searchText !== nextValue) {
      this.props.router.push(`/materials/search/${nextValue}`)
    }
  }

  viewMaterial = (idMaterial) => {
    this.props.router.push(`/materials/${idMaterial}`)
  }


  render() {
    const { children, showFilter, filters, visibleMaterials, locale, loading } = this.props
    const searchText = this.props.params.searchText || ''
    let gallery
    if (loading || !searchText) {
      gallery = null
    } else {
      gallery = visibleMaterials.length > 0
        ? React.cloneElement(children, { key: 'materialList', materials: visibleMaterials, locale, viewMaterial: this.viewMaterial })
        : <p>{<FormattedMessage {...messages.materialsNotFound} />}</p>
    }
    // const gallery = visibleMaterials.length > 0 ? React.cloneElement(children, { materials: visibleMaterials, locale, viewMaterial: this.viewMaterial }) : null
    // this code in return is not so good if children changes (search, categories...)
    //  {visibleMaterials.length > 0 && <MaterialList materials={visibleMaterials} />}
    return (
      <View>
        <Helmet
          title='PictogramsView'
          meta={[
            { name: 'description', content: 'Description of PictogramsView' }
          ]}
        />
        <Toggle
          label={<FormattedMessage {...messages.advancedSearch} />}
          onToggle={this.props.toggleShowFilter}
          defaultToggled={showFilter}
          style={{ width: '200px', float: 'right' }}
        />
        <SearchField value={searchText} onSubmit={this.handleSubmit} />
        {showFilter ? <FilterList filtersMap={filters} setFilterItems={this.props.setFilterItems} filtersData={this.props.filtersData} /> : null}
        {gallery}
      </View>
    )
  }
}

MaterialsView.propTypes = {
  requestMaterials: PropTypes.func.isRequired,
  toggleShowFilter: PropTypes.func.isRequired,
  searchText: PropTypes.string,
  loading: PropTypes.bool.isRequired,
  params: PropTypes.object.isRequired,
  filters: PropTypes.instanceOf(Map),
  filtersData: PropTypes.instanceOf(Map),
  showFilter: PropTypes.bool,
  setFilterItems: PropTypes.func.isRequired,
  visibleMaterials: PropTypes.arrayOf(PropTypes.object),
  // Injected by React Router
  children: PropTypes.node,
  router: PropTypes.any.isRequired,
  locale: PropTypes.string.isRequired
}


const mapStateToProps = (state, ownProps) => {
  const filters = state.getIn(['materialsView', 'filters'])
  const showFilter = state.getIn(['materialsView', 'showFilter'])
  const locale = state.get('language').get('locale')
  const loading = state.getIn(['materialsView', 'loading'])
  const filtersData = state.getIn(['configuration', 'filtersData'])
  // const activeFilters = state.getIn(['materialsView', 'filters'])
  // denormalize: https://github.com/paularmstrong/normalizr/blob/master/docs/api.md#denormalizeinput-schema-entities
  // const materialSchema = new schema.Entity('materials', {}, { idAttribute: 'idMaterial' })
  // const searchMaterialSchema = [materialSchema]
  const searchResults = state.getIn(['materialsView', 'search', locale, ownProps.params.searchText]) || []
  const entities = {}
  entities.materials = state.getIn(['materialsView', 'materials']).toJS()
  const materialList = denormalize(searchResults, searchMaterialSchema, entities)
  const visibleMaterials = getFilteredItems(materialList, filters.toJS())
  // const visibleMaterials = materialList

  return ({
    filters,
    showFilter,
    visibleMaterials,
    locale,
    loading,
    filtersData
  })
}

const mapDispatchToProps = (dispatch) => ({
  requestMaterials: (locale, searchText) => {
    dispatch(materials.request(locale, searchText))
  },
  toggleShowFilter: () => {
    dispatch(toggleShowFilter())
  },
  setFilterItems: (filter, filterItem) => {
    dispatch(setFilterItems(filter, filterItem))
  }
})

export default connect(mapStateToProps, mapDispatchToProps)(withRouter(MaterialsView))
