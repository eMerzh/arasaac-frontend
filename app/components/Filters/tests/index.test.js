import React from 'react'
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider'
import { setFilterItems } from 'containers/MaterialsView/actions'
import createComponentWithIntl from 'utils/createComponentWithIntl'
import { Map } from 'immutable'
import Component from '../index'

jest.mock('material-ui/internal/Tooltip')
jest.mock('../FilterSelectLoader')

describe('<FilterList />', () => {
  const filtersData = Map({
    activity: [{ 1: 'act1' }, { 2: 'act2' }, { 3: 'act3' }],
    language: [{ es: 'spanish' }, { it: 'italian' }],
    area: [{ 1: 'area1' }, { 2: 'area2' }, { 3: 'area3' }]
  })
  const filtersMap = Map({
    activity: [1, 2],
    language: ['es'],
    area: [1]
  })

  it('renders correctly', () => {
    const tree = createComponentWithIntl(
      <MuiThemeProvider>
        <Component setFilterItems={setFilterItems} filtersMap={filtersMap} filtersData={filtersData} />
      </MuiThemeProvider>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
