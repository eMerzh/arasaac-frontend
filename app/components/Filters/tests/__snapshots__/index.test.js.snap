exports[`<FilterList /> renders correctly 1`] = `
<div
  style={
    Object {
      "marginBottom": "3rem",
    }
  } />
`;
